extmod/vfs_fat_diskio.c

```diff
@@ -180,7 +180,7 @@ DRESULT disk_write (
         if (nlr_push(&nlr) == 0) {
             mp_obj_t ret = mp_call_method_n_kw(2, 0, vfs->writeblocks);
             nlr_pop();
-            if (mp_obj_get_int(ret) != 0) {
+            if (ret != mp_const_none && MP_OBJ_SMALL_INT_VALUE(ret) != 0) {
```

Comment:
`!mp_obj_is_true(ret)` will make it use the standard Python truthiness. https://github.com/adafruit/circuitpython/blob/master/py/obj.h#L681

Reply:
The particular test I used above was already used elsewhere in this file (see line 73, for example), so let's stick with the current test. The implicit contract is that the called routines either have a plain `return` (so returning `None`, meaning everything is fine, or return an error code, with `0` meaning OK. Other Falsiness (like empty lists) is not expected.